docs_src/design/adr/0027-URIs for Files.md

```diff
@@ -0,0 +1,72 @@
+
+### Authentication 
+
+While not recommended, users will be able to specify **basic-auth** (`<username>:<password>@`) in the URI in plain text. In order to provide a secure way for users to specify credentials, the URI scheme will be expanded to allow specifying a Secret Name from the service's Secret Store. This will take the form  `[secretName: <secret-name>]@`. The type of authentication as well as the credentials will be contained in the secret data specified by the Secret Name. Three types of authentication will be supported, which are `usernamepassword`, `apikey` and `bearertoken`. 
+
+- When `usernamepassword` is specified the `[secretName: <secret-name>]@` text in the URI will be replaced with `username:password@` using the **username** and **password** found in the secret data.
+
+    !!! example - "Example secret data - `usernamepassword`"
+        ```
+        type=usernamepassword
+        username=myuser
+        password=mypassword
+        ```
+
+- When `apikey` is specified, the `[secretName: <secret-name>]@` text will be removed from the URI and the **API Key** will be placed in the HTTP header using `apikeyname` and `apikeyvalue`  from the secret data
+
+    !!! example - "Example secret data - `apikey`"
+        ```
+        type=apikey
+        apikeyvalue=mykeyvalue
+        apikeyname=myname
+        ```
+    
+- When `bearertoken` is specified, the `[secretName: <secret-name>]@` text will be removed from the URI and the **Bearer Token** will be placed in the HTTP header as  `Authorization` with value of `Bearer <token>`  where `token` is from the secret data. 
+
+    !!! example - "Example secret data - `token`"
+        ```
+        type=bearertoken
+        token=mytoken
+        ```
+
+### Services/Files Impacted
+
+- **All Services** will be impacted for enabling the loading the **common configuration** and **private configuration** files using URIs. This will be handled in `go-mod-bootstrap's` processing of the `-cc/--commonConfig` and `-cf/--configFile` command line flags.
+
+- **Core Metadata's** loading of the **UOM file** will be adjusted to use the new file load function.
+
+- **Device Service's** loading of **device profiles**, **device definitions** and **provision watches** files will be adjusted to load an index file specified by a URI in place of the configured folder name. The contents of the index file will be used to load the individual files by URI  by appending the filenames to the original URI. Any authentication specified in the original URI will be used in the subsequent URIs. 
+
+    !!! example - "Example Device Index file http://example.com/devices/index.json"
+        ```json
+        [
+            "device1.yaml", "device2.yaml"
+        ]
+        ```

```

Comment:
This example needs to be completed by showing the resulting URI. I'm not getting the picture.

Reply:
Ok

Reply:
added